### DISCUSSION TITLE
Add config to disable reputation or modify reputation globally

### DISCUSSION THREAD

**Discussion by Phoupraw** [ejektaflex/Bountiful#288] [Mod Ideas]
The reputation is shared across all the players, so the discount is increasing much faster than in singleplayer. I like the bounty board system but don't want the reputation to ease those tasks.

**Comment by ejektaflex** [comment]
I can look into adding a config option to disable Reputation for boards. It's impossible to make reputation be per-player, since reputation affects bounties generated on the board and what all players see on the board is the same. :^)

**Reply by tamanor** [reply]
I was really hoping there would be a way in the future it make it so it was per-player :/. 
I wanted to build a guild hall and and these setup, but on a server it a shame that the boards are globally synced to everyone. 

hmmm, is there no way you could maybe do a system kind of like how enchanting tables work based on the players xp. 
Like could you make it so if a player does not have a certain level of reputation the higher level bounties are greyed out like they do on the enchanting table?. 
that way everyone would see the same bounties still but only the people with the right reputation can accept them?

**Reply by ejektaflex** [reply]
They're globally synced, but they're still per-board. Perhaps you could create areas that only each player can enter? That way everyone has their own separate board.

In any case, having per-player-per-board bounties would mean that there are suddenly a lot of extra bounties to keep track of. If someone had a server with 100 players, then there could be potentially 100*24 = 2400 bounties on a single board. It would also be generating bounties 100x as often per board. I don't like to consider the performance implications of syncing the NBT data of a single bounty board on a server of 100 players with this method - to support this kind of optimization, the complexity of the project would go up, and it would be harder to maintain. Which could be worthwhile, but I do not know many people who would use it - as far as I can tell, people prefer to all see the same bounties so that they can tell their friends when a good one shows up.

I might consider looking into making reputation a per-player thing, but only after 1.21 when I can evaluate where we are. The component changes in 1.21 potentially mean that network transfer of data is a bit more performant!

**Reply by tamanor** [reply]
I technically you could still have it so the boards are globally synced right? and have it so there is a separate reputation stat that is counted per player and if said players do not have the correct reputation level the high level bounties would be greyed out kind of like how the enchanting table does it. 

But in terms of coding and getting something like that up, I'm not sure how difficult that would be. ^^

Edit: Then again, This would mean that each board would have to separate character levels so I can see how that would be more harder to do
